MoreCyclopsUpgrades/SaveData/ModConfig.cs

```diff
@@ -0,0 +1,224 @@
+                ValidDataRead &= false;
+            }
+
+            if (!EmUpgradesEnabled.HasValue)
+            {
+                QuickLogger.Warning($"Config value for {ConfigKey}>{EmUpgradesEnabled.Key} was out of range. Replaced with default.");
+                ValidDataRead &= false;
+            }
+
+            if (!EmBioEnergyEnabled.HasValue)
+            {
+                QuickLogger.Warning($"Config value for {ConfigKey}>{EmBioEnergyEnabled.Key} was out of range. Replaced with default.");
+                ValidDataRead &= false;
+            }
+        }
+
+        internal override EmProperty Copy()
+        {
+            return new ModConfig();
+        }
+
+        private void WriteConfigFile()
+        {
+
+            File.WriteAllLines(SaveFile, new[]
+            {
+                versionLine,
+                PrettyPrint(),
+                "",
+                "# Here's the full details on what these configurations do: #",
+                "",
+                "# 'Enable Auxiliary Upgrade Consoles' #",
+                "# When this option is enabled, the new Auxiliary Cyclops Upgrade Console will be patched into the game. #",
+                "# This is an upgrade console you can build inside your cyclops to give yourself an extra six upgrade slots. #",
+                "# Set this to 'NO' if you want the added challenge of having to manage with only six upgrade slots. #",
+                "# Set this to 'YES' if you want to be able to equip every cyclops upgrade module you ever come across. #",
+                "",
+                "# 'Enable New Upgrade Modules' #",
+                "# When this option is enabled, all seven upgrade modules created for this mod will be patched into the game. #",
+                "# Set this to 'NO' if you want a closer to vanilla experience. #",
+                "# Set this to 'YES' if you want to unleash the full potential of your cyclops. #",
+                "",
+                "# 'Enable Cyclops Bioreactor' #",
+                "# When this option is enabled, the Cyclops Bioreactor will be patched into the game. #",
+                "# Set this to 'NO' if you want a closer to vanilla experience. #",
+                "# Set this to 'YES' if you want additional power options for your Cyclops. #",
+                "",
+                "# 'Cyclops Power Level' #",
+                "# This setting lets you configure the overall balance of the entire mod. #",
+                "# If you find that the Cyclops is too easy to too hard to maintain, try changing this setting. #",
+                "# Changing this will limit the number of charging modules that can be added to the Cyclops of any one type. #",
+               $"# {CyclopsPowerLevels.Leviathan} (Easy Mode) #",
+               $"# {CyclopsPowerLevels.Ampeel} (Modest Difficulty) #",
+               $"# {CyclopsPowerLevels.Crabsnake} (Moderate Difficulty) #",
+               $"# {CyclopsPowerLevels.Peeper} (Hard Mode) #",
```

Comment:
Make sure this gets properly documented and informs exactly what is being changed.